### PR TITLE
Feat/ replace opentracing by opentelemetry

### PR DESCRIPTION
**Pull request by manuellysuzik**
Changes made:
- The Tracer Server Interceptor now is handled by openTelemetry instead of openTracing.
- The Tracer also was replaced from one created by OpenTracing to the OpenTelemetry module.

### CODE REVIEW

**Comment by bitbanshee** [comment]
In order to make traces reporting vendor agnostic, it would be nice to use the [Open Telemetry exporter](https://opentelemetry.io/docs/instrumentation/go/exporting_data/#otlp-exporter) instead of Jeager's.

Jaeger all-in-one deployment can still be used for integration tests, although it is necessary to expose the port 4317 that is the entrypoint for Open Telemetry trace reporting via gRPC. [See the table detailing ports in Jaeger's documentation.](https://www.jaegertracing.io/docs/next-release/getting-started/#all-in-one)

**Comment by manuellysuzik** [comment]
> In order to make traces reporting vendor agnostic, it would be nice to use the [Open Telemetry exporter](https://opentelemetry.io/docs/instrumentation/go/exporting_data/#otlp-exporter) instead of Jeager's.
> 
> Jaeger all-in-one deployment can still be used for integration tests, although it is necessary to expose the port 4317 that is the entrypoint for Open Telemetry trace reporting via gRPC. [See the table detailing ports in Jaeger's documentation.](https://www.jaegertracing.io/docs/next-release/getting-started/#all-in-one)

PTAL AGAIN?
Captura de Tela 2022-08-10 às 19 03 47

File: app/app.go [hunk 1]
```diff
@@ -106,14 +112,29 @@ func (a *App) configure() error {
 	return nil
 }
 
-func (a *App) configureJaeger() error {
-	opts := jaeger.Options{
-		Disabled:    a.config.GetBool("jaeger.disabled"),
-		Probability: a.config.GetFloat64("jaeger.samplingProbability"),
-		ServiceName: a.config.GetString("jaeger.serviceName"),
+func (a *App) configureOTEL(ctx context.Context) error {
```
Comment (lmsilva-wls) [inline app/app.go]: You don't need to receive a `context.Context` here

also nit: I think the capitalization should be `OTel` instead of `OTEL`

**Review by lmsilva-wls — CHANGES_REQUESTED** [review]
Looks good to me overall, I left one comment; by the way, I did not see any configuration of the OTEL exporter

File: app/app.go (hunk 1 above)
Comment (manuellysuzik) [reply app/app.go]: You are right! PTAL again?

**Review by lmsilva-wls — APPROVED** [review]

File: go.mod [hunk 2]
```diff
@@ -22,24 +22,34 @@ require (
...
 	github.com/topfreegames/go-extensions-kafka v1.1.0
 	github.com/topfreegames/protos v1.6.1
-	google.golang.org/grpc v1.31.0
+	go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc v0.34.0
+	go.opentelemetry.io/otel v1.9.0
+	google.golang.org/grpc v1.48.0
+)
+
+require (
+	go.opentelemetry.io/otel/exporters/jaeger v1.9.0
```
Comment (bitbanshee) [inline go.mod]: I believe we don't need this anymore, right?

**Review by bitbanshee — CHANGES_REQUESTED** [review]

File: go.mod (hunk 2 above)
Comment (manuellysuzik) [reply go.mod]: Done!

**Pull request merged**